src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/AzureADStartup.cs

```diff
@@ -0,0 +1,60 @@
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Microsoft.Identity.Web;
+using OrchardCore.Deployment;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.Microsoft.Authentication.Configuration;
+using OrchardCore.Microsoft.Authentication.Deployment;
+using OrchardCore.Microsoft.Authentication.Drivers;
+using OrchardCore.Microsoft.Authentication.Recipes;
+using OrchardCore.Microsoft.Authentication.Services;
+using OrchardCore.Microsoft.Authentication.Settings;
+using OrchardCore.Modules;
+using OrchardCore.Navigation;
+using OrchardCore.Recipes;
+using OrchardCore.Security.Permissions;
+
+namespace OrchardCore.Microsoft.Authentication;
+
+[Feature(MicrosoftAuthenticationConstants.Features.AAD)]
+public sealed class AzureADStartup : StartupBase
+{
+    public override void ConfigureServices(IServiceCollection services)
+    {
+        services.TryAddEnumerable(new ServiceDescriptor(typeof(IPermissionProvider), typeof(Permissions), ServiceLifetime.Scoped));
```

Comment:
I know this was copied, but here you may want to consider using `AddPermissionProvider<>` extension.

```suggestion
        services.AddPermissionProvider<Permissions>();
```